### PR TITLE
nanoemoji: init at 0.15.1

### PR DESCRIPTION
**Pull request by 999eagle**
###### Description of changes

This adds nanoemoji, see https://github.com/googlefonts/nanoemoji.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/nanoemoji/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    toml
+    tomlkit
+    ufo2ft
+    ufoLib2
+    zopfli
+  ];
+
+  nativeCheckInputs = with python3.pkgs; [
+    pytestCheckHook
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/nanoemoji/default.nix:52]:
```suggestion
```
included in pytestCheckHook

File: pkgs/development/python-modules/picosvg/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  propagatedBuildInputs = [
+    absl-py
+    lxml
+    skia-pathops
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/picosvg/default.nix:46]:
```suggestion
```

File: pkgs/development/python-modules/ninja/default.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest
+    pytest-cov
+    pytest-runner
+    pytest-virtualenv
+    codecov
+    coverage
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ninja/default.nix:59]:
```suggestion
    pytest-virtualenv
```
runner is useless nad we don't want to collect coverage

File: pkgs/development/python-modules/ninja/default.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    rev = version;
+    hash = "sha256-scCYsSEyN+u3qZhNhWYqHpJCl+JVJJbKz+T34gOXGJM=";
+  };
+  patches = [
+    # make sure cmake doesn't try to download the ninja sources
+    ./no-download.patch
+  ];
+
+  # see NinjaUrls.cmake
+  ninja_src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ninja/default.nix:33]: Should we assert that the version matches?

File: pkgs/development/python-modules/ninja/default.nix (hunk 3 above)
Comment (999eagle) [reply pkgs/development/python-modules/ninja/default.nix]: `pytest-cov` is required due to the `setup.cfg`. I've removed the rest.

File: pkgs/development/python-modules/ninja/default.nix (hunk 4 above)
Comment (999eagle) [reply pkgs/development/python-modules/ninja/default.nix]: I've added a check based on the hash

File: pkgs/development/python-modules/ninja/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/ninja/default.nix]: Just remove the options from setup.cfg, we do that all the time

Comment (SuperSandro2000) [reply pkgs/development/python-modules/ninja/default.nix]: Just remove the options from setup.cfg, we do that all the time

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: pkgs/tools/misc/nanoemoji/default.nix [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    picosvg
+  ];
+
+  makeWrapperArgs = [
+    "--prefix PATH : ${lib.makeBinPath [ pngquant resvg ]}"
+  ];
+
+  preCheck = ''
+    # make sure the built binaries (nanoemoji/maximum_color) can be found by the test
+    export PATH="$out/bin:''${PATH}"
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/nanoemoji/default.nix:63]:
```suggestion
    export PATH="$out/bin:$PATH"
```
we can avoid the quoting hell like this

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**